### PR TITLE
Feature/vru two heavy vehicles

### PR DESCRIPTION
**Pull request by EricChen-Lei**
# PR Details
## Description
add 3 trucks for VRU scenario

## Related Jira Key
CDAR-432

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/bus_stop.add.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!-- generated on 2023-11-06 15:52:06 by Eclipse SUMO netedit Version 1.18.0
+-->
+
+<additional xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://sumo.dlr.de/xsd/additional_file.xsd">
+    <!-- StoppingPlaces -->
+    <busStop id="bs_2" lane="10_3" startPos="2.00" endPos="8.00"/>
+</additional>
```
Comment (MishkaMN) [inline co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/bus_stop.add.xml:9]: If we are not using bus stop, do we still need this? or is also being used for the heavy_vehicles to stop?

**Review by MishkaMN — DISMISSED** [review]

File: co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.rou.xml [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
     <!-- Vehicles, persons and containers (sorted by depart) -->
-    <vehicle id="v_0" depart="999.00" route="r_0"/>
-</routes>
\ No newline at end of file
+    <vehicle id="v_0" type="vehicle.carlamotors.carlacola" depart="0.00" route="r_2"/>
+    <vehicle id="v_1" type="vehicle.carlamotors.carlacola" depart="3.00" route="r_1"/>
+    <vehicle id="v_2" type="vehicle.carlamotors.carlacola" depart="8.00" route="r_0">
+    	<stop busStop="bs_2" duration="5.00"/>
+    </vehicle>
+    <vehicle id="v_3" type="vehicle.carlamotors.carlacola" depart="999.00" route="r_0"/>
```
Comment (MishkaMN) [inline co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.rou.xml:17]: what is this one? seems like there are 4 vehicles?

Comment (EricChen-Lei) [reply co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.rou.xml:17]: This one is used for keeping the scenario alive, if we only have three cars, the simulation will stop after the last car finishes, might not give us time for other Carma stuff we want to do, this is a safety

File: co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/bus_stop.add.xml (hunk 1 above)
Comment (EricChen-Lei) [reply co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/bus_stop.add.xml:9]: Yes, it's actually used in the rou.xml file, under the third car

File: co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.sumocfg [hunk 3]
```diff
@@ -5,13 +5,12 @@
...
-        <route-files value="Town04.rou.xml"/>
-        <additional-files value="detector.xml"/>
-        <gui-settings-file value="gui.settings.xml"/>
-
+        <route-files value="carlavtypes.rou.xml,Town04.rou.xml"/>
+        <additional-files value="detector.xml, bus_stop.add.xml"/>
     </input>
     <gui_only>
         <tracker-interval value="0.1"/>
+        <gui-settings-file value="viewsettings.xml"/>
```
Comment (paulbourelly999) [inline co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.sumocfg:13]: Why did this change?

File: co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.sumocfg [hunk 4]
```diff
@@ -5,13 +5,12 @@
     </traci_server>
     <input>
         <net-file value="Town04.net.xml"/>
-        <route-files value="Town04.rou.xml"/>
-        <additional-files value="detector.xml"/>
-        <gui-settings-file value="gui.settings.xml"/>
```
Comment (paulbourelly999) [inline co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.sumocfg:10]: Why did we remove this. This file is required for setting the simulation speed for SUMO

File: co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.sumocfg (hunk 3 above)
Comment (EricChen-Lei) [reply co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.sumocfg:14]: This is for a different purpose, this is for representing vehicle types in the sumo gui

File: co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.sumocfg (hunk 4 above)
Comment (EricChen-Lei) [reply co-simulation/bundle/src/assembly/resources/scenarios/Town04/sumo/Town04.sumocfg:10]: This was taken out by accident, added back in

**Review by MishkaMN — APPROVED** [review]

**Pull request merged**